From f9bcfed5a1d44d9211c5f6eba403a9898c8c9057 Mon Sep 17 00:00:00 2001
From: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
Date: Tue, 8 Aug 2023 19:03:13 +0100
Subject: [PATCH] kea: fix reproducible build failure

New version of Kea has started using path of build-dir instead of
src-dir which results in reproducible builds failure.
Use src-dir as is used in v2.2.0

Upstream-Status: Pending
https://gitlab.isc.org/isc-projects/kea/-/issues/3007

Upstream has confirmed the patch will not be accepted but discussions
with upstream is still going on, we might have a proper solution later.

Signed-off-by: Sudip Mukherjee <sudipm.mukherjee@gmail.com>
---
 src/bin/admin/kea-admin.in | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/bin/admin/kea-admin.in b/src/bin/admin/kea-admin.in
index 034a0ee..8ab11ab 100644
--- a/src/bin/admin/kea-admin.in
+++ b/src/bin/admin/kea-admin.in
@@ -51,14 +51,14 @@ dump_qry=""
 if test -f "@datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh"; then
     . "@datarootdir@/@PACKAGE_NAME@/scripts/admin-utils.sh"
 else
-    . "@abs_top_builddir@/src/bin/admin/admin-utils.sh"
+    . "@abs_top_srcdir@/src/bin/admin/admin-utils.sh"
 fi
 
 # Find the installed kea-lfc if available. Fallback to sources otherwise.
 if test -x "@sbindir@/kea-lfc"; then
     kea_lfc="@sbindir@/kea-lfc"
 else
-    kea_lfc="@abs_top_builddir@/src/bin/lfc/kea-lfc"
+    kea_lfc="@abs_top_srcdir@/src/bin/lfc/kea-lfc"
 fi
 
 # Prints out usage version.
@@ -355,7 +355,7 @@ mysql_upgrade() {
     # Check if there are any files in it
     num_files=$(find "${upgrade_scripts_dir}" -name 'upgrade*.sh' -type f | wc -l)
     if [ "$num_files" -eq 0 ]; then
-        upgrade_scripts_dir=@abs_top_builddir@/src/share/database/scripts/mysql
+        upgrade_scripts_dir=@abs_top_srcdir@/src/share/database/scripts/mysql
 
         # Check if the scripts directory exists at all.
         if [ ! -d ${upgrade_scripts_dir} ]; then
@@ -405,7 +405,7 @@ pgsql_upgrade() {
     # Check if there are any files in it
     num_files=$(find "${upgrade_scripts_dir}" -name 'upgrade*.sh' -type f | wc -l)
     if [ "$num_files" -eq 0 ]; then
-        upgrade_scripts_dir=@abs_top_builddir@/src/share/database/scripts/pgsql
+        upgrade_scripts_dir=@abs_top_srcdir@/src/share/database/scripts/pgsql
 
         # Check if the scripts directory exists at all.
         if [ ! -d ${upgrade_scripts_dir} ]; then
-- 
2.39.2

